synapse/handlers/e2e_keys.py

```diff
@@ -247,6 +254,15 @@ async def do_remote_query(destination: str) -> None:
                         user_results = results.setdefault(user_id, {})
                         for device in user_devices:
                             user_results[device["device_id"]] = device["keys"]
+
+                        # update the result with the devicelist's cross-signing keys
+                        cross_signing_keys["master_keys"][user_id] = user_devices.get(
+                            "master_key"
+                        )
+                        cross_signing_keys["self_signing_keys"][
+                            user_id
+                        ] = user_devices.get("self_signing_key")
```

Comment:
this will overwrite any existing values in `cross_signing_keys`. Is that correct behaviour?